### PR TITLE
Output for generators

### PR DESCRIPTION
**Pull request by ShakoHo**
@askeing @MikeLien @ypwalter r?

### CODE REVIEW

File: lib/common/commonUtil.py [hunk 1]
```diff
@@ -1,31 +1,50 @@
...
-        self.current_status = load_json_file(status_fp)
+        self.current_status = CommonUtil.load_json_file(status_fp)
+
+    def get_current_status(self):
+        return self.current_status['current_status']
+
+    def record_current_status(self, input_status_dict):
+        self.current_status = CommonUtil.load_json_file(self.status_fp)
+        if 'current_status' in self.current_status:
+            self.current_status['current_status'].update(input_status_dict)
```
Comment (ypwalter) [inline lib/common/commonUtil.py]: The same question you came up last time. Should there be a transitional variable for this update()?

File: lib/generator/frameThroughputDctGenerator.py [hunk 2]
```diff
@@ -190,30 +193,77 @@ def crop_images_based_on_samplefiles(self, input_data):
...
+                           'threshold': self.index_config.get('compare-threshold', 0.0003),
+                           'search_margin': self.index_config.get('search-margin', 10)}
+        self.compare_result['running_time_result'] = compare_with_sample_image_multi_process(
+            input_data['sample_result'],
+            input_image_list,
+            compare_setting)
         # get frame throughput values
-        compare_result.update(
-            self.get_frame_throughput(compare_result['running_time_result'], compare_result['merged_crop_image_list']))
+        if self.compare_result is not None:
```
Comment (MikeLien) [inline lib/generator/frameThroughputDctGenerator.py]: This line will be a logic implicit error, `self.compare_result` will always not equal to `None` even it's only equal to `{}`.
I guess here should to check the running time result exists or not to decide following calculation is needed or not, right?

File: lib/generator/inputLatencyAnimationDctGenerator.py [hunk 3]
```diff
@@ -91,28 +95,68 @@ def crop_images_based_on_samplefiles(self, input_data):
...
-
-        return compare_result
+            'threshold': self.index_config.get('compare-threshold', 0.0003),
+            'search_margin': self.index_config.get('search-margin', 10)}
+        self.compare_result['running_time_result'] = compare_with_sample_image_multi_process(
+            input_data['sample_result'],
+            input_image_list,
+            compare_setting)
+
+        if self.compare_result is not None:
```
Comment (MikeLien) [inline lib/generator/inputLatencyAnimationDctGenerator.py]: same as `lib/generator/frameThroughputDctGenerator.py` comment

File: lib/generator/runTimeDctGenerator.py [hunk 4]
```diff
@@ -113,23 +120,85 @@ def crop_images_based_on_samplefiles(self, input_data):
...
+                           'generator_name': self.__class__.__name__,
+                           'skip_status_bar_fraction': self.SKIP_STATUS_BAR_FRACTION,
+                           'exec_timestamp_list': input_data['exec_timestamp_list'],
+                           'threshold': self.index_config.get('compare-threshold', 0.0003),
+                           'search_margin': self.index_config.get('search-margin', 10)}
+        self.compare_result['running_time_result'] = compare_with_sample_image_multi_process(
+            input_data['sample_result'],
+            input_image_list,
+            compare_setting)
+        if self.compare_result is not None:
```
Comment (MikeLien) [inline lib/generator/runTimeDctGenerator.py]: same as `lib/generator/frameThroughputDctGenerator.py` comment

File: lib/generator/inputLatencyAnimationDctGenerator.py [hunk 5]
```diff
@@ -91,28 +95,68 @@ def crop_images_based_on_samplefiles(self, input_data):
...
+            run_time, event_time_dict = CalculationUtil.runtime_calculation_event_point_base(self.compare_result['running_time_result'])
+            self.compare_result.update({'run_time': run_time, 'event_time_dict': event_time_dict})
+        return self.compare_result
+
+    def output_case_result(self, suite_upload_dp):
+
+        if self.compare_result is not None:
```
Comment (ypwalter) [inline lib/generator/inputLatencyAnimationDctGenerator.py]: same as @MikeLien's comment

File: lib/generator/runTimeDctGenerator.py [hunk 6]
```diff
@@ -113,23 +120,85 @@ def crop_images_based_on_samplefiles(self, input_data):
...
+            if self.index_config['calculate-speed-index']:
+                si_progress = calculate_progress_for_si(self.compare_result['running_time_result'],
+                                                        self.compare_result['merged_crop_image_list'])
+                self.compare_result['speed_index'] = calculate_speed_index(si_progress)
+                self.compare_result['perceptual_speed_index'] = calculate_perceptual_speed_index(si_progress)
+
+        return self.compare_result
+
+    def output_case_result(self, suite_upload_dp):
+        if self.compare_result is not None:
```
Comment (ypwalter) [inline lib/generator/runTimeDctGenerator.py]: same as @MikeLien's comment

File: lib/common/commonUtil.py (hunk 1 above)
Comment (ShakoHo) [inline lib/common/commonUtil.py]: Not sure what you mean, could you be more specific?

Comment (ypwalter) [inline lib/common/commonUtil.py]: https://github.com/Mozilla-TWQA/Hasal/pull/626/files/1f61c38933bf71dee47ce3793e6e2ca8a7d92d37#diff-84a84bffe5e405f666440db9f8213141R144

When we discussed about .update() last time, I think we should be using a transitional variable before .update()?

File: lib/common/commonUtil.py [hunk 7]
```diff
@@ -36,12 +55,199 @@ def record_status(self, case_name, status, value):
             json.dump(self.current_status, fh)
 
 
+class CalculationUtil(object):
```
Comment (askeing) [inline lib/common/commonUtil.py]: Should we re-write the `outlier` base on `CalculationUtil`?

Ref:
https://github.com/Mozilla-TWQA/Hasal/blob/dev/lib/common/outlier.py#L5
https://github.com/Mozilla-TWQA/Hasal/blob/dev/lib/helper/resultHelper.py#L128-L130
https://github.com/Mozilla-TWQA/Hasal/blob/dev/lib/helper/generatorHelper.py#L188-L190
https://github.com/Mozilla-TWQA/Hasal/blob/dev/server/server.py#L666

**Pull request merged**